src/object.c

```diff
@@ -377,6 +377,66 @@ void decrRefCount(robj *o) {
     }
 }
 
+void unusedStringObject(robj *o) {
+    if (o->encoding == OBJ_ENCODING_RAW) {
+        redisUnusedMemory(sdsAllocPtr(o->ptr));
+    }
+}
+
+void unusedSetObject(robj *o) {
+    switch (o->encoding) {
+    case OBJ_ENCODING_HT:
+        redisUnusedMemory(((dict*)o->ptr)->ht[0].table);
+        redisUnusedMemory(((dict*)o->ptr)->ht[1].table);
+        break;
```

Comment:
the members in a set (and fields in hash, or list) can be big (in some cases).
did we decide we're not gonna handle them?

if we're afraid of the extra iteration, maybe we wanna integrate that freeing logic into the saving code, i.e. release list nodes and hash fields one by one when iterating on them for saving.

considering today we have preamble AOF, and it's by default, i'm even willing to overlook the old AOFRW code and only do that for rdb saving.

Reply:
yes, i am afraid of the extra iteration, i am not sure it is right way to integrate that freeing logic into the saving code, that may make code coupled.

I think it is usual that AOF only stores redis command, so i did that.